### PR TITLE
knightos-kcc: init at 4.0.0

### PR DESCRIPTION
**Pull request by siraben**
###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [x] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Review by danieldk — CHANGES_REQUESTED** [review]
Thanks! I have added some suggested improvements.

**Comment by siraben** [comment]
I think it's ready to merge.

File: pkgs/development/compilers/kcc/default.nix [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+  version = "4.0.0";
+
+  src = fetchFromGitHub {
+    owner = "KnightOS";
+    repo = "kcc";
+    rev = version;
+    sha256 = "1cd226nqbxq32mppkljavq1kb74jqfqns9r7fskszr42hbygynk4";
+  };
+
+  nativeBuildInputs = [ cmake boost bison flex ];
```
Comment (danieldk) [inline pkgs/development/compilers/kcc/default.nix:15]:
```suggestion
  nativeBuildInputs = [ cmake bison flex ];

  buildInputs = [ boost ];
```

File: pkgs/development/compilers/kcc/default.nix [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+
+  meta = with stdenv.lib; {
+    homepage    = "https://knightos.org/";
+    description = "KnightOS C compiler";
+    license     = licenses.gpl2;
```
Comment (danieldk) [inline pkgs/development/compilers/kcc/default.nix:20]:
```suggestion
    license     = licenses.gpl2Plus;
```

`gpl2` deprecated, use `gpl2Only` or `gpl2Plus`. From a quick look at the source code, they use the 'or later wording', so this should be `gpl2Plus`.

**Review by danieldk — CHANGES_REQUESTED** [review]

**Review by danieldk — APPROVED** [review]
LGTM

Result of `nixpkgs-review pr 95891` [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
<br>- knightos-kcc
</details>

**Pull request merged**